src/coord/src/coord.rs

```diff
@@ -3362,6 +3359,12 @@ impl Coordinator {
                                                 return Err(CoordError::Unstructured(anyhow!(e)))
                                             }
                                         };
+                                        let (name, typ) = desc.name_typ_at_index(*idx);
```

Comment:
Is it possible to have the update and insert code paths share the data validation code so that we don't hit this class of bug again? This will be useful if we ever support CHECK constraints.